src/apps/testapps/testH3UniEdge.c

```diff
@@ -208,34 +209,113 @@ TEST(getH3UnidirectionalEdgesFromHexagon) {
 }
 
 TEST(getH3UnidirectionalEdgeBoundary) {
-    H3Index sf = H3_EXPORT(geoToH3)(&sfGeo, 9);
+    H3Index sf;
+    GeoBoundary boundary;
+    GeoBoundary edgeBoundary;
     H3Index* edges = calloc(6, sizeof(H3Index));
-    H3_EXPORT(getH3UnidirectionalEdgesFromHexagon)(sf, edges);
 
-    GeoBoundary gb;
-    for (int i = 0; i < 6; i++) {
-        H3_EXPORT(getH3UnidirectionalEdgeBoundary)(edges[i], &gb);
-        t_assert(gb.numVerts == 2, "Got the expected number of vertices back");
+    int expectedVertices[][2] = {{3, 4}, {1, 2}, {2, 3},
+                                 {5, 0}, {4, 5}, {0, 1}};
+
+    for (int res = 0; res < 13; res++) {
```

Comment:
Note that the current approach fails at res > 12 because of the geo comparison precision.

Reply:
Might be worth adding as a comment.

Reply:
Will do.